### PR TITLE
Bug fix/unlink view on smart edge collection

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

When we created a view on a SmartEdgeCollection, the internal handling on the
index disallowed to unlink the view again.

- [x] :hankey: Bugfix 
- [ ] :pizza: New feature 
- [ ] :hammer: Refactoring 
- [x] :book: CHANGELOG entry made
- [ ] :muscle: The behavior in this PR was *manually tested*
- [ ] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [ ] No backports required
- [x] Backports required for: *(Please specify versions)*
  - [ ] 3.7
  - [ ] 3.6
  - [ ] 3.5
#### Related Information

*(Please reference tickets / specification etc)*

- [ ] Docs PR: 
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/570
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [x] This PR adds tests that were used to verify all changes:
  - [x] Added **Regression Tests**
  - [ ] Added new C++ **Unit Tests**
  - [x] Added new **integration tests** (i.e. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)

Additionally:

- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12451/

### Documentation

> All new Features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the changelog so that
> developers and users have a concise overview. 

- [ ] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)
- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: arangod/IResearch/IResearchLinkHelper.cpp [hunk 1]
```diff
@@ -187,13 +189,18 @@ arangodb::Result dropLink( // drop link
...
-    arangodb::iresearch::IResearchLink const& link // link to drop
+template <>
+arangodb::Result dropLink<arangodb::iresearch::IResearchViewCoordinator>(  // drop link
+    arangodb::LogicalCollection& collection,        // link collection
+    arangodb::iresearch::IResearchLink const& link  // link to drop
 ) {
-  arangodb::velocypack::Builder builder;
+  // Enterprise variant, we only need to drop links on non-hidden
+  // collections (e.g. in SmartGraph Case)
+  if (arangodb::ClusterMethods::filterHiddenCollections(collection)) {
```
Comment (gnusi) [inline arangod/IResearch/IResearchLinkHelper.cpp:199]: how we can handle link properties changes then?

**Review by gnusi — CHANGES_REQUESTED** [review]

File: arangod/IResearch/IResearchLinkHelper.cpp (hunk 1 above)
Comment (mchacki) [reply arangod/IResearch/IResearchLinkHelper.cpp:199]: Discussed offline, as this is mainly closed Enterprise Code.

File: arangod/IResearch/IResearchLinkHelper.cpp [hunk 2]
```diff
@@ -136,9 +137,14 @@ arangodb::Result createLink( // create link
     arangodb::iresearch::IResearchViewCoordinator const& view, // link view
     arangodb::velocypack::Slice definition // link definition
 ) {
+  if (arangodb::ClusterMethods::filterHiddenCollections(collection)) {
+    // Silently ignore hidden collections
```
Comment (gnusi) [inline arangod/IResearch/IResearchLinkHelper.cpp:141]: let's at least log this fact

**Review by gnusi — CHANGES_REQUESTED** [review]

File: arangod/IResearch/IResearchLinkHelper.cpp (hunk 2 above)
Comment (mchacki) [reply arangod/IResearch/IResearchLinkHelper.cpp]: Discussed in offline call

**Comment by mchacki** [comment]
Jenkins:
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12462/

Green besides known issue in GO test.

**Comment by mchacki** [comment]
After test modification:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12473/

**Review by gnusi — APPROVED** [review]

**Comment by mchacki** [comment]
Jenkins now blue:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12478/

**Pull request merged**